src/_use_stable_memo/index.jsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useRef, useState } from 'react';
+
+const useStableMemo = (fn, teardownFn, deps) => {
+  const firstRun = useRef(true);
+  const [memorized, setMemorized] = useState(() => fn());
+
+  useEffect(() => {
+    let localMemorized;
+
+    if (firstRun.current) {
+      firstRun.current = false;
+      localMemorized = memorized;
+    } else {
+      localMemorized = fn();
+      setMemorized(() => localMemorized);
```

Comment:
you don't need state, storing in ref should work just fine

Reply:
I am pretty sure I do. One test case would even fail without the state here.

I need to cause a new render cycle to get the new value out to the component using this hook. If you return keep memorized in a ref then you cannot use it as an effect dep.